doc/user/flowspec.rst

```diff
@@ -0,0 +1,330 @@
+implement this policy routing.
+
+- by filtering the traffic based on `Linux NetFilter`.
+
+`Linux NetFilter` provides a set of tools like ipset and iptables that are powerful
+enough to be able to filter such flowspec filter rule.
+
+- by using non standard routing tables (through ip rule tool from `Linux Iproute2`).
+
+That tool is already used by pbrd daemon that provides basic routing based on IP source,
+and IP destination criterium.
+
+Below example is an illustration of what BGP flowspec will inject in the underlying
+system:
+
+.. code-block:: shell
+
+   # linux shell
+   ipset create match0x102 hash:net,net counters
+   ipset add match0x102 32.0.0.0/16,40.0.0.0/16
+   iptables -N match0x102 -t mangle
+   iptables -A match0x102 -t mangle -j MARK --set-mark 102
+   iptables -A match0x102 -t mangle -j ACCEPT
+   iptables -i ntfp3 -t mangle -I PREROUTING -m set --match-set match0x102 src,dst
+                -g match0x102
+   ip rule add fwmark 102 lookup 102
+   ip route add 40.0.0.0/16 via 44.0.0.2 table 102
+
+For handling an incoming Flowspec entry, the following workflow is applied:
+
+- incoming Flowspec entries are handled by `bgpd`, and are stored in the BGP RIB.
+- Flowspec entry will be installed according to its complexity.
+
+It will be installed if one of the following filtering action is seen on the BGP
+extended community: either redirect IP, or redirect VRF, in conjunction with rate
+option, for redirecting traffic. Or rate option set to 0, for discarding traffic.
+
+According to the degree of complexity of the Flowspec entry, it will be installed
+in *zebra* RIB. For more information about what is supported in the FRR implementation
+as rule, see `Limitations / Known Issues` chapter. Flowspec entry is split inseveral
+parts before being sent to *zebra*.
+
+- *zebra* daemon is receiving the policy routing configuration
+
+Policy-routing entities necessary to policy route the traffic in the underlying system,
+is received by *zebra*. Two filtering contexts will be created or appended (`Linux
+NetFilter ipset` and `Linux netfilter iptable` entry). If the traffic has to be dropped,
+then those filtering contexts will be enough. Otherwise, the traffic will be redirected
+to an other routing table, with in that routing table, a route entry to redirect the
+traffic to the wished destination.
+
+Configuration guide
+-------------------
+
+In order to configure an IPv4 flowspec engine, use the following configuration. As of
+today, it is only possible to configure Flowspec on default VRF.
+
+.. code-block:: frr
+
+   router bgp <AS>
+     neighbor <A.B.C.D> remote-as <remoteAS>
+     address-family ipv4 flowspec
+      neighbor <A.B.C.D> activate
+    exit
+   exit
+
+You can see Flowspec entries, by using one of the following show commands:
+
+.. index:: show bgp ipv4 flowspec [detail | A.B.C.D]
+.. clicmd:: show bgp ipv4 flowspec [detail | A.B.C.D]
+
+
+Flowspec Per Interface
```

Comment:
I would replace this with `Per-interface configuration`, since this is under the `Flowspec` subsection of the docs.

Reply:
ok